src/test/java/com/uber/cadence/workflow/WorkflowTest.java

```diff
@@ -1147,10 +1147,17 @@ private void waitForProc(WorkflowExecution execution) {
     workflowClient.newUntypedWorkflowStub(execution, Optional.empty()).getResult(Void.class);
   }
 
+  private Map<String, Object> getTestMemo() {
+    Map<String, Object> memo = new HashMap<>();
+    memo.put("testKey", "testObject");
+    return memo;
+  }
+
   @Test
   public void testStart() {
     startWorkerFor(TestMultiargsWorkflowsImpl.class);
-    WorkflowOptions workflowOptions = newWorkflowOptionsBuilder(taskList).build();
+    WorkflowOptions workflowOptions =
+        newWorkflowOptionsBuilder(taskList).setMemo(getTestMemo()).build();
```

Comment:
Add check (may be using the WorkflowExecutionStarted event) that the memo was actually passed to the service.

Reply:
Done